exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/ProofMapIndexProxyIntegrationTest.java

```diff
@@ -681,27 +681,6 @@ void clearFailsIfSnapshot() {
     });
   }
 
-  /**
-   * A simple integration test that ensures that:
-   * - ProofMap constructor preserves the index type and
-   * - Map constructor checks it, preventing illegal access to ProofMap internals.
-   */
-  @Test
-  void constructorShallPreserveTypeInformation() {
```

Comment:
@vitvakatu , @skletsun 

Now we test that in `indexConstructorPersistsIndexTypeInfo` for _each_ index :rocket: It is disabled, however, as with de-duplication requires a working "merge" operation.